### ISSUE TITLE
Checksum mismatch for glog 0.3.5

### ISSUE THREAD

**Issue by muellermartin** [Homebrew/homebrew-core#13495]
Running `brew upgrade` fails for Formula glog 0.3.5 when compiling it from source due to SHA256 checksum mismatch.

Here is the full output:
```
==> Upgrading glog 
rm /usr/local/include/glog
rm /usr/local/lib/libglog.0.dylib
rm /usr/local/lib/libglog.a
rm /usr/local/lib/libglog.dylib
rm /usr/local/lib/pkgconfig/libglog.pc
rm /usr/local/share/doc/glog-0.3.4
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20170512-16929-m5oho9.sb nice /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/glog.rb --verbose
==> Downloading https://github.com/google/glog/archive/v0.3.5.tar.gz
Already downloaded: /Users/martin/Library/Caches/Homebrew/glog-0.3.5.tar.gz
==> Verifying glog-0.3.5.tar.gz checksum
ln -s ../Cellar/glog/0.3.4_1/include/glog glog
ln -s ../../Cellar/glog/0.3.4_1/share/doc/glog-0.3.4 glog-0.3.4
ln -s ../Cellar/glog/0.3.4_1/lib/libglog.0.dylib libglog.0.dylib
ln -s ../Cellar/glog/0.3.4_1/lib/libglog.a libglog.a
ln -s ../Cellar/glog/0.3.4_1/lib/libglog.dylib libglog.dylib
ln -s ../../Cellar/glog/0.3.4_1/lib/pkgconfig/libglog.pc libglog.pc
Error: SHA256 mismatch
Expected: 277846fa6ac3d569fed48c8e32191ffd286ca52b808f243c81a4a91a9f9ff113
Actual: 7580e408a2c0b5a89ca214739978ce6ff480b5e7d8d7698a2aa92fadc484d1e0
Archive: /Users/martin/Library/Caches/Homebrew/glog-0.3.5.tar.gz
To retry an incomplete download, remove the file above.
```

Removing and redownloading the offending file as suggested did not help.

The error might likely be introduced by PR #13395 by @ilovezfs.

Temporary workaround: `brew pin glog`.

**Comment by ilovezfs** [comment]
@muellermartin this means they were either hacked or retagged, as #13395 went through our CI successfully. Either way, someone needs to contact upstream to find out exactly what happened.

**Comment by muellermartin** [comment]
@ilovezfs The release for 0.3.5 at https://github.com/google/glog/releases/tag/v0.3.5 currently states it was "released this 2 days ago" but your PR #13395 is from "3 days ago" therefore I would sey they retagged their release...

**Comment by ilovezfs** [comment]
@muellermartin Regardless, they'll need to be contacted to find out exactly what happened.

**Comment by muellermartin** [comment]
If you look at the commit log you see what happened:

<img width="993" alt="github-glog-commit-log-0 3 5" src="https://cloud.githubusercontent.com/assets/343302/26008299/763f1724-3744-11e7-8b18-af657c2b0a1b.png">

**Comment by MikeMcQuaid** [comment]
@muellermartin Contacting upstream is Homebrew policy so we need to do so, regardless.

**Comment by muellermartin** [comment]
¯\_(ツ)_/¯

**Comment by lorenzb** [comment]
I contacted upstream here: 
https://github.com/google/glog/issues/189

**Comment by ilovezfs** [comment]
@lorenzb thank you for contacting upstream. The re-tag has been confirmed: https://github.com/google/glog/issues/189#issuecomment-301305664

**Comment by ilovezfs** [comment]
Closed via #13559.
